core/ordersync/ordersync.go

```diff
@@ -300,30 +306,47 @@ func (s *Service) GetOrders(ctx context.Context, minPeers int) error {
 			log.WithFields(log.Fields{
 				"provider": peerID.Pretty(),
 			}).Trace("requesting orders from neighbor via ordersync")
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			default:
-			}
 
-			if err := s.getOrdersFromPeer(ctx, peerID); err != nil {
-				log.WithFields(log.Fields{
-					"error":    err.Error(),
-					"provider": peerID.Pretty(),
-				}).Warn("could not get orders from peer via ordersync")
-				continue
-			} else {
-				// TODO(albrow): Handle case where no orders were returned from this
-				// peer. This could be considered a valid response, depending on the implementation
-				// details of the subprotocol. We need to not try them again, but also not count
-				// them toward the number of peers we have successfully synced with.
-				log.WithFields(log.Fields{
-					"provider": peerID.Pretty(),
-				}).Trace("succesfully got orders from peer via ordersync")
-				successfullySyncedPeers.Add(peerID.Pretty())
-			}
+			wg.Add(1)
+			go func(id peer.ID) {
+				select {
```

Comment:
Usually I like to include `defer wg.Done()` as the very first line of the goroutine to make sure we don't forget it and it always runs regardless of code path. Is there a reason to not do that here? Failing to call `wg.Done` even once will cause deadlock.